File: src/test/java/compiler/Parameter.java
package compiler;

public class Parameter {

    private String type;
    private String name;

    // Represents what is input for ASM. E.g. int = 'I', String = 'Ljava/lang/String;'
    private String asmType;

    public Parameter(String type, String name) {
        this.type = type;
        this.name = name;


    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }

    @Override
    public String toString() {
        return type + " : " + name;
    }

    public String getAsmType() {

        if (type.equals("int")) {
            return "I";
        }
        if (type.equals("String")) {
            return "Ljava/lang/String;";
        }
        return null;
    }

    public void setAsmType(String asmType) {
        this.asmType = asmType;
    }
}

File: src/main/scala/compiler/structure/parameters/Parameter.scala
package compiler.structure.parameters

// Represents what is input for ASM. E.g. int = 'I', String = 'Ljava/lang/String;'
class Parameter(var `type`: String, var name: String) {



  // Name of the variable
  def getName: String = name

  // ASM type. int = "I", String = "java.lang.String", etc
  def getAsmType: String = {
    if (getType == "int") "I"

    else if (getType == "String") "Ljava/lang/String;"

    else null
  }

  // Type of the variable
  def getType: String = `type`

  def setType(`type`: String) {
    this.`type` = `type`
  }

  override def toString: String = {
    return name + " : " + `type`
  }

}
